Sitecore/developer-portal
# Initial Sitecore CDP/P Integration

## Description / Motivation

Adding in tracking support for CDP/P.  For lower environments we'll use a test tenant for testing and then production will use Sitecore.com tenant.  The goal of this is to add MVP level tracking, and then we'll likely enhance this in future projects.

- In additional I have added Environment variables to Vercel.  All non-production environments should point to the `DemoTenant1 (US)` in the `Sitecore Dev Relations` org in Cloud portal.  Production is pointing to the Sitecore.com tenant (which is currently not available on cloud portal).
- Currently will only track page view events, as a first phase of this.  This does not track search terms or anything else, and we can add these in incrementally once this PR is complete.
- This includes updates to the product tagging in Front matter, since the page view will track the url as well as the product if it's available.

## Testing Tracking

To test that it's tracking correctly, make sure you've configured the .env.local environment variables for CDP locally (or they are already configured for CDP/P in Vercel environments).  If they are not, in the console, you will see a warning about configuration issue.  Follow the following instructions in a non-production Vercel instance, to confirm that it's tracking events correctly:

1. Go to the site and open up the developer tools and access the console and type the following: `Engage.getBrowserId()`.  If you get an error, than there is something wrong.  Otherwise you should get back a guid.
2. Copy the Guid and go to the `DemoTenant1 (US)` Personalize tenant, and access the Event Viewer. 
3. in the search bar, type in: `bid: {guid}` and it should show you another Id.  This is the Guest id associated with the Browser Id you grabbed from the browser.
4. If you click the Guest Id, you should now see all events for that guest, and now you should be able to navigate around the site and confirm it's adequately tracking page views.

## Types of changes

- [ ] Documentation update (non-breaking change; modified files are limited to the `/data` directory or other markdown files)

## Checklist:




- [X] My code/comments/docs fully adhere to the Code of Conduct.
- [X] My change is a code change.
- [ ] My change is a documentation change and there are NO other updates required.
- [ ] My change has new or updated images which are stored in the `/public/images` folder that need to be migrated to Sitecore DAM
